cmd/util/app_test.go

```diff
@@ -225,6 +234,48 @@ func Test_setAppSpecOptions(t *testing.T) {
 	})
 }
 
+func newMultiSourceAppOptionsFixture() *appOptionsFixture {
+	fixture := &appOptionsFixture{
+		spec: &v1alpha1.ApplicationSpec{
+			Sources: v1alpha1.ApplicationSources{
+				v1alpha1.ApplicationSource{},
+				v1alpha1.ApplicationSource{},
+			},
+		},
+		command: &cobra.Command{},
+		options: &AppOptions{},
+	}
+	AddAppFlags(fixture.command, fixture.options)
+	return fixture
+}
+
+func Test_setAppSpecOptionsMultiSourceApp(t *testing.T) {
```

Comment:
Can you add a test case when index is specified as 0, Would it return an error ?

Reply:
The function if specified index as 0, would take the source at index 0. Added a test for that.